### ISSUE TITLE
Add Password

### ISSUE THREAD

**Issue by Quantum-Future** [nain-F49FF806/sharepaste.oo#15]
Good app. !
Is it possible to add support for setting a password?

**Comment by nain-F49FF806** [comment]
> Good app. !

Thank you. Glad to hear you are liking it. 

> Is it possible to add support for setting a password?

It is. 

But its utility is very limited, I believe. And correctly using it is actually tricky. So it has not been a priority. 

To understand your use case, can you answer a few questions?

1. What is the purpose of the password?
2. How would you securely share the password?

3. Are you aware that we already generate a secure encryption key, 
  and that a password won't increase the encryption strength?

**Comment by Quantum-Future** [comment]
Well, thinking about your questions, I realized that you're right: the utility is limited.
I'm aware of point 3: a password won't increase the encription stength, just thougth it will increase security by adding an additional "layer" (the need to know a password), but maybe it's not much worthwhile.

**Comment by nain-F49FF806** [comment]
> .. thougth it will increase security by adding an additional "layer" (the need to know a password), but maybe it's not much worthwhile

Yes, passwords in relation to privatebin work on the concept of [preshared key](https://en.wikipedia.org/wiki/Pre-shared_key). You must *already* have agreed to a password with the person you want to communicate with, like in person, or over phone. 

In such a scenario, password could be useful. However, in such a scenario, you might as well coordinate a more secure and reliable way to communicate, like Signal messenger / [Autocrypt email](https://delta.chat/). 


To improve practical security with strangers (i.e no pre-agreed password), you could use the "burn on read" feature. This will ensure that the paste expires as soon as your recipient opens it, preventing anyone else from being able to access it thereafter. This of course assumes your communication isn't actively intercepted, but at most passively monitored.
